src/tail/tail.rs

```diff
@@ -346,7 +346,18 @@ fn follow<T: Read>(readers: &mut [BufReader<T>], filenames: &[String], settings:
     let mut read_some = false;
     let mut process = platform::ProcessChecker::new(settings.pid);
 
+    let mut buffer = BufReader::new(stdin());
     loop {
+        loop {
+            let mut datum = String::new();
+            match buffer.read_line(&mut datum) {
+                Ok(0) => break,
+                Ok(_) => {
+                  print!("input consumed: {}", datum);
```

Comment:
What do you mean by "does nothing"? I can't quite believe the behaviour of this loop will change if you make that line say `OK(_) => {},`. The docs would certainly mention an oddity like this, and they don't.

I think you can rewrite the whole `match` like this:

```rust
if buffer.read_line(&mut datum).unwrap() == 0 {
    break;
}
```